### PR TITLE
chore: migrate to vitest

### PR DESCRIPTION
**Pull request by jrasm91**
* 20 seconds vs 40+ seconds for unit tests
* coverage doesn't seem to work for some reason
* run via swc
* remove dependency on jest-when

### CODE REVIEW

File: server/src/domain/trash/trash.service.spec.ts [hunk 1]
```diff
@@ -8,16 +8,17 @@ import {
   newCommunicationRepositoryMock,
   newJobRepositoryMock,
 } from '@test';
+import { Mocked } from 'vitest';
 import { JobName } from '..';
```
Comment (danieldietzler) [inline server/src/domain/trash/trash.service.spec.ts:12]: Not your fault but a bad import.

File: server/test/repositories/communication.repository.mock.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-export const newCommunicationRepositoryMock = (): jest.Mocked<ICommunicationRepository> => {
+export const newCommunicationRepositoryMock = (): Mocked<ICommunicationRepository> => {
   return {
-    send: jest.fn(),
-    broadcast: jest.fn(),
-    on: jest.fn(),
-    sendServerEvent: jest.fn(),
+    send: vi.fn(),
+    broadcast: vi.fn(),
+    on: vi.fn() as any,
```
Comment (danieldietzler) [inline server/test/repositories/communication.repository.mock.ts:8]: Why this `as any`?

**Review by danieldietzler — APPROVED** [review]
Nice!

File: server/test/repositories/communication.repository.mock.ts (hunk 2 above)
Comment (jrasm91) [reply server/test/repositories/communication.repository.mock.ts:8]: this method has multiple signatures and it was throwing a type error.

File: server/package.json [hunk 3]
```diff
@@ -123,42 +124,9 @@
     "ts-node": "^10.9.1",
```
Comment (benmccann) [inline server/package.json:124]: probably don't need this anymore

Comment (benmccann) [reply server/package.json:124]: also `@types/jest` above

Comment (jrasm91) [reply server/package.json:124]: It is still used in e2e, but I may try to migrate those here in this PR as well

**Pull request merged**